### ISSUE TITLE
Building for Android 28 cmake error

### ISSUE THREAD

**Issue by rakybond007** [pocl/pocl#817]
Hello, I am trying to build pocl for Android 28 arm64-v8a.
Environment 
1. **Using llvm-config**: /home/hj/ndk/**android-ndk-r20b**/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-config
2. **Built llvm-8 libraries for Android architecture**. (llvm-config's LLVM_LIBS is: -lLLVMCoverage -lLLVMDlltoolDriver -lLLVMLineEditor -lLLVMOrcJIT -lLLVMWindowsManifest -lLLVMFuzzMutate -lLLVMMCJIT -lLLVMCoroutines -lLLVMLTO -lLLVMPasses -lLLVMObjCARCOpts -lLLVMipo...)
3. Used **Android-sdk cmake** : android-sdk/cmake/3.10.2.4988404/bin/cmake
4. There were no llc, opt binaries for android on ndk toolchain, but seems to use host's binaries. -> I copied from Desktop's bin to Android toolchain directory.

With Environment above, I tried to build pocl with the command
/home/hj/ndk/android-sdk/cmake/3.10.2.4988404/bin/cmake -DCMAKE_TOOLCHAIN_FILE=/home/hj/ndk/android-sdk/ndk/21.0.6113669/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DANDROID_PLATFORM=28 -DLLVM_DEFAULT_TARGET_TRIPLE=aarch64-linux-android -DWITH_LLVM_CONFIG=/home/hj/ndk/android-ndk-r20b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-config ..

But I get the Error message of 
CMakeFiles/cmTC_0515c.dir/build.make:97: recipe for target 'cmTC_0515c' failed
CMake Error at cmake/LLVM.cmake:768 (message):
  LLVM link test FAILED.  This mostly happens when your LLVM installation
  does not have all dependencies installed.
Call Stack (most recent call first):
  CMakeLists.txt:442 (include)

I saw a developer who uploaded Pocl-prebuilt for ARM32 android before. Please help me to build or fix the error. Thanks

**Comment by FreemanX** [comment]
The error message tells you to check LLVM.cmake.  `CMake Error at cmake/LLVM.cmake:768 (message)` Maybe you can check what's going on there. Passing all checks and configurations defined in LLVM.cmake can be one of the most troublesome steps in the whole process. There'll be more errors coming from LLVM.cmake later. Understanding what LLVM.cmake does helps a lot.

**Comment by rakybond007** [comment]
Thanks for @FreemanX , I got a lot help and finally built LLVM and other libraries to build PoCL!

Now I started to build the PoCL, but I got some error which seems to be related with My target architecturem GalaxyS10 aarch64-linux-android with TARGET_CPU : **cortex-a55**

However, make fails soon with those error message.
![image](https://user-images.githubusercontent.com/5442998/80944270-17f32900-8e24-11ea-98f6-7da6462b9d45.png)
I cannot guess what is the role of the file, what can I check to solve the problem. Can someone give me some Intuition? Thanks :)
